google/cloud/firestore_v1/base_document.py

```diff
@@ -315,10 +315,10 @@ def _prep_collections(
 
     def collections(
         self, page_size: int = None, retry: retries.Retry = None, timeout: float = None,
-    ) -> NoReturn:
```

Comment:
Is there a difference in these returns? Seems unrelated to the change otherwise?

Reply:
There is! `NoReturn` tells IDEs that the function _never returns_, so any code appearing after such a method call is deemed unreachable. To that end, I cleaned up any `NoReturn` type hints I found on methods that my code was calling.

Reply:
One interesting detail is that you may be referencing is that this parent class implementation does nothing, so this method does actually _never return_. That said, other classes that can accept either `Document` or `AsyncDocument` instances might specify their type hints with `BaseDocument`, and in those methods, IDEs think any code that appears after calling `some_document.collections()` is unreachable.

Reply:
I was sincerely curious, not secretly knowing the answer here :) Thanks for explaining!

Reply:
I only learned this myself working on BulkWriter when I finally got fed up with VS Code "getting confused" about certain code paths being unreachable. Like always, it was just PEBKAC.